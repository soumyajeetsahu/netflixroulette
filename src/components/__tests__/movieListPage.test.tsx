import MovieListPage from '../movieList/movieListPage';
import React from 'react';
import { render, fireEvent, screen, waitForElementToBeRemoved } from '@testing-library/react';
import '@testing-library/jest-dom';
import axios, { AxiosResponse } from 'axios';
import userEvent from '@testing-library/user-event';

describe('MovieListPage', () => {
    afterEach(() => {
        jest.restoreAllMocks();
    });
    beforeEach(() => {
        const mAxiosResponse = {
            data: {
                data: [
                    {
                        "id": 337167,
                        "title": "Fifty Shades Freed",
                        "tagline": "Don't miss the climax",
                        "vote_average": 6.1,
                        "vote_count": 1195,
                        "release_date": "2018-02-07",
                        "poster_path": "https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg",
                        "overview": "Believing they have left behind shadowy figures from their past, newlyweds Christian and Ana fully embrace an inextricable connection and shared life of luxury. But just as she steps into her role as Mrs. Grey and he relaxes into an unfamiliar stability, new threats could jeopardize their happy ending before it even begins.",
                        "budget": 55000000,
                        "revenue": 136906000,
                        "genres": [
                            "Drama",
                            "Romance"
                        ],
                        "runtime": 106
                    }]
            }
            ,
        } as AxiosResponse;
        jest.spyOn(axios, 'get')
            .mockResolvedValueOnce(mAxiosResponse);
    })


    test('renders Header and ListMovies components when showMovieDetails is false', () => {
        render(<MovieListPage />);

        expect(screen.getByText('FIND YOUR MOVIE')).toBeInTheDocument();
        const input = screen.getByPlaceholderText('What do you want to watch?');

        expect(input).toBeInTheDocument();
    });

    test('renders MovieDetailsComponent when movie tile is clicked', async () => {
        render(<MovieListPage />);

        expect(await screen.findByText('Fifty Shades Freed')).toBeInTheDocument();
        const movieTile = screen.getByText('Fifty Shades Freed');
        fireEvent.click(movieTile);
        await waitForElementToBeRemoved(() => screen.getByText('FIND YOUR MOVIE'));
        expect(screen.getByTestId('SearchIcon')).toBeInTheDocument();
    });

    test('hides MovieDetailsComponent and shows Header when backToSearch is called', async () => {
        render(<MovieListPage />);

        expect(await screen.findByText('Fifty Shades Freed')).toBeInTheDocument();
        const movieTile = screen.getByText('Fifty Shades Freed');
        fireEvent.click(movieTile);
        await waitForElementToBeRemoved(() => screen.getByText('FIND YOUR MOVIE'));
        expect(screen.getByTestId('SearchIcon')).toBeInTheDocument();
        const backButton = screen.getByTestId('SearchIcon');
        fireEvent.click(backButton);

        expect(await screen.findByText('Fifty Shades Freed')).toBeInTheDocument();
    });

    test('calls "onSelect" callback with correct genre after clicking a genre button', async () => {
        render(<MovieListPage />);
        expect(await screen.findByText('Fifty Shades Freed')).toBeInTheDocument();
        const genreButton = screen.getByText('Documentary');
        fireEvent.click(genreButton);
        const selectedGenre = screen.getByText('Documentary');
        expect(selectedGenre.closest('a')).toHaveClass('active');
    });

    test('calls "onSelect" callback with correct genre after clicking a genre button', async () => {
        render(<MovieListPage />);
        expect(await screen.findByText('Fifty Shades Freed')).toBeInTheDocument();
        const genreButton = screen.getByText('Documentary');
        fireEvent.click(genreButton);
        const selectedGenre = screen.getByText('Documentary');
        expect(selectedGenre.closest('a')).toHaveClass('active');
    });

    test('calls the onSearch function with the entered search term', async () => {
        const component = render(<MovieListPage/>);
        expect(await screen.findByText('Fifty Shades Freed')).toBeInTheDocument();
        fireEvent.change(screen.getByPlaceholderText("What do you want to watch?"), {
          target: { value: 'Fifty Shades Freed' },
        });
    
        fireEvent.click(screen.getByText("Search"));
        expect(component.baseElement).toMatchSnapshot();
      });
});