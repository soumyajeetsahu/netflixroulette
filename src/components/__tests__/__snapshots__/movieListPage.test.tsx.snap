// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListPage calls the onSearch function with the entered search term 1`] = `
<body>
  <div>
    <div
      class="homeContainer"
    >
      <div
        class="headerDiv"
      >
        <div
          class="pageTitle"
        >
          netflixroulette
        </div>
        <div
          class="addButtonDiv"
        >
          <button
            class="addButton"
          >
            + add movie
          </button>
        </div>
        <div
          class="pageLabel"
        >
          FIND YOUR MOVIE
        </div>
        <div
          class="searchForm"
        >
          <form
            class="searchForm"
          >
            <div
              class="search"
            >
              <input
                class="searchInput"
                placeholder="What do you want to watch?"
                type="text"
                value=""
              />
              <button
                class="searchButton"
                type="submit"
              >
                Search
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="listMovies-Container"
      >
        <div
          class="genre-container"
        >
          <div
            class="genres-line"
          >
            <a
              class="genres active"
            >
              <span>
                ALL
              </span>
            </a>
            <a
              class="genres "
            >
              <span>
                Documentary
              </span>
            </a>
            <a
              class="genres "
            >
              <span>
                Comedy
              </span>
            </a>
            <a
              class="genres "
            >
              <span>
                Horror
              </span>
            </a>
            <a
              class="genres "
            >
              <span>
                Crime
              </span>
            </a>
          </div>
        </div>
        <div
          class="sortControl"
        >
          <span
            class="sortByText"
          >
            SORT BY
          </span>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-30jyav-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          >
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              role="button"
              tabindex="0"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </div>
            <input
              aria-hidden="true"
              aria-invalid="false"
              class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
              tabindex="-1"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-ihdtdm"
              >
                <span
                  class="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="movieTiles"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-vkqe12-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiBox-root css-79elbk"
          >
            <img
              alt="Fifty Shades Freed"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-og4kvs-MuiCardMedia-root"
              height="380"
              src="https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg"
            />
            <button
              aria-label="settings"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1vohci7-MuiButtonBase-root-MuiIconButton-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-3ms82c-MuiSvgIcon-root"
                data-testid="MoreVertIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 8c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm0 2c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2zm0 6c-1.1 0-2 .9-2 2s.9 2 2 2 2-.9 2-2-.9-2-2-2z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="MuiCardContent-root css-bp185q-MuiCardContent-root"
          >
            <div
              class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
            >
              Fifty Shades Freed
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeSmall MuiButton-outlinedSizeSmall MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeSmall MuiButton-outlinedSizeSmall css-neo7gc-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
               
              2018
            </button>
          </div>
          <div
            class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
          >
            <span
              class="MuiTypography-root MuiTypography-caption css-1tykgd6-MuiTypography-root"
            >
              <span>
                Drama
                , 
              </span>
              <span>
                Romance
              </span>
            </span>
          </div>
        </div>
      </div>
      <div
        style="margin-top: 1em; color: rgb(246, 82, 97); font-family: Montserrat; font-size: 20px;"
      >
        netflixroulette
      </div>
    </div>
  </div>
</body>
`;
